imagedephi/redact/redact.py

```diff
@@ -59,31 +58,18 @@ def redact_images(
     base_rules = get_base_rules()
     images_to_redact = iter_image_files(input_path) if input_path.is_dir() else [input_path]
     redact_dir = create_redact_dir(output_dir)
+    show_redaction_plan(input_path)
     for image_file in images_to_redact:
-        if image_file.suffix not in FILE_EXTENSION_MAP:
-            click.echo(f"Image format for {image_file.name} not supported. Skipping...")
-            continue
-        try:
-            redaction_plan = build_redaction_plan(image_file, base_rules, override_rules)
-        except tifftools.TifftoolsError:
-            click.echo(f"Could not open {image_file.name} as a tiff. Skipping...")
-            continue
-        except MalformedAperioFileError:
-            click.echo(
-                f"{image_file.name} could not be processed as a valid Aperio file. Skipping..."
-            )
-            continue
-        except UnsupportedFileTypeError as e:
-            click.echo(f"{image_file.name} could not be processed. {e.args[0]}")
-            continue
         click.echo(f"Redacting {image_file.name}...")
-        if not redaction_plan.is_comprehensive():
-            click.echo(f"Redaction could not be performed for {image_file.name}.")
-            redaction_plan.report_missing_rules()
-        else:
-            redaction_plan.execute_plan()
-            output_path = _get_output_path(image_file, redact_dir)
-            redaction_plan.save(output_path, overwrite)
+        if image_file.suffix in FILE_EXTENSION_MAP:
+            redaction_plan = build_redaction_plan(image_file, base_rules, override_rules)
+            if not redaction_plan.is_comprehensive():
+                click.echo(f"Redaction could not be performed for {image_file.name}.")
+                redaction_plan.report_missing_rules()
+            else:
+                redaction_plan.execute_plan()
+                output_path = _get_output_path(image_file, redact_dir)
+                redaction_plan.save(output_path, overwrite)
 
```

Comment:
@naglepuff just wanted to point out this refactor. I removed these checks since we are now calling `show_redaction_plan` when `run` is called

Reply:
I like having the checks in only one place. My only concern is that exceptions can still be raised in `build_redaction_plan`, so they should be handled in some other way here. Maybe there's a way to keep track of which files raised an exception in `show_redaction_plan`?

Once we have a more robust way of file type checking, I think we can get rid of a lot of these checks.

Reply:
I _think_ these concerns are handled in a combination of the addition of using the magic bits to identify tiffs and also the pending logging PR. If I am wrong let me know and I'll be happy to find a suitable solution.